### ISSUE TITLE
[Core] Improve the logging around worker exit type

### ISSUE THREAD

**Issue by yncxcw** [ray-project/ray#16212]
### Describe your feature request

We have seen the workers in our production cluster got killed for no reason(no sign of any error). After days of debugging, we finally located that this is related to `TryKillingIdleWorkers`, a mechanism added after Ray 1.0.

We hope we can have more information around exit type in logs, right now the log like this is of no use for debugging.
```
Exit signal received, this process will exit after all outstanding  tasks have finished, exit_type=INTENDED_EXIT.
```

I have a PR illustrate the idea: https://github.com/ray-project/ray/pull/16211

For feature requests or questions, post on our Discussion page instead: https://discuss.ray.io/

**Comment by rkooo567** [comment]
Hmm actor shoudn't be killed by `TryKillingIdleWorkers`. This only kills idle "workers" (so actors should never be killed). What version of ray are you using?

**Comment by yncxcw** [comment]
Sorry, it is actually workers. Rephrase it to workers. We are right now at Ray-1.3.0. The issue we are seeing is that it takes quite a long time for the workers to be ready if we request a large number of workers.

**Comment by rkooo567** [comment]
Btw, this is invoked only when there are more actors than num_cpus (so if you have like 40 workers in 32 cpu machine, which is not ideal). Can you explain a little bit more about your workload and a problem? (Btw, I am not against your change. I am trying to understand if we can fix the underlying issues here).

**Comment by yncxcw** [comment]
Yeah, so we are having 60 cores, but we are requesting like 200 actors for each node. The logic is kind of like this:
```python
@ray.remote
class Loader:

    def __call__():
        # do something


s = time.time()   
loaders = [Loader.remote() for _ in range(200)]
sync_across_nodes() # We use smoething like horovod to sync across all drirvers on all nodes.
e = time.time()
# once all actors are up for everying node, we do something
```
For Ray 1.0, `e-s` <= 2mins
Starting from Ray 1.2, `e-s` >= 30 mins

Our solution is to set `kill_idle_workers_interval_ms` in ray.init() to 0.

**Comment by rkooo567** [comment]
I see. I think the most native solution is

```
@ray.remote(num_cpus=0.25)
class Loader
```

This way, you can have 60 * 4 workers per node without worrying about `TryKillingIdleWorkers `. Turning that feature flag also   could work. 

(Btw, note that num_cpus should be cleanly divide 1, for example, if you set num_cpus=0.33, there could be some issues due to how we manage resources)

**Comment by yncxcw** [comment]
Thanks, yeah, we can try this.

The argument I made in the PR is that it would be really nice if we can make the exit type more explicit so users know what's going on when debugging things like this.
